server/src/Api.hs

```diff
@@ -115,6 +114,8 @@ appServer env = hoistServer api appHandler server
             throwError err400 {errBody = LC8.pack ih}
           OtherDecodeError str ->
             throwError err400 {errBody = LC8.pack str}
+        handleError (ErrorCall err) =
+          throwError err400 {errBody = LC8.pack $ show err}
```

Comment:
Curious why we're not returning Json errors by simply having an instance for `CtlServerError`?

Reply:
These are only for the users (they are not recoverable), thus they don't have to be in a machine-readable format

Reply:
That's ok, a few cents tho...

Generally speaking, you don't make logs and error messages machine readable just because they are recoverable, you make it so monitoring and analytics stack can parse them out easily. That's why a combination of syslog+machine readable format for logs and error messages are a good thing to introduce early on.

For instance, CTL server is just one runtime component in our infrastructure, we generally want to introduce log collection and analysis for all these services, this whole process is predicated on the assumption that collected information can be easily parsed.